test/e2e/performanceprofile/functests/1_performance/cpu_management.go

```diff
@@ -329,35 +306,41 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", Ordered, func() {
 			Expect(err).ToNot(HaveOccurred())
 
 			By("Getting the CPU scheduling flags")
-			flags, err := getCPUsSchedulingDomainFlags()
+			// After the testpod is started get the schedstat and check for cpus
+			// not participating in scheduling domains
+			cpusNotInSchedulingDomains, err := getCPUsSchedulingDomains()
 			Expect(err).ToNot(HaveOccurred())
 
 			By("Verifying that the CPU load balancing was disabled")
 			for _, cpu := range cpus.ToSlice() {
-				Expect(len(flags[cpu])).To(Equal(len(defaultFlags[cpu])))
-				// the CPU flags should be almost the same except the LSB that should be disabled
-				// see https://github.com/torvalds/linux/blob/0fe5f9ca223573167c4c4156903d751d2c8e160e/include/linux/sched/topology.h#L14
-				// for more information regarding the sched domain flags
-				for i := range flags[cpu] {
-					Expect(flags[cpu][i]).To(Equal(defaultFlags[cpu][i] - 1))
+				for _, str := range cpusNotInSchedulingDomains {
+					if strings.Contains(str, fmt.Sprint(cpu)) {
+						cpuFound = true
+						break
+					}
 				}
 			}
+			Expect(cpuFound).To(BeTrue())
```

Comment:
I think you want to check all cpus here, not just the first one that is disabled.

Reply:
But we are interested only in the cpus used by the gu pod has load balancing disabled right ?

Reply:
Oh, I missed the outer loop over `cpus`. Why use a double loop here? Can't you return a map from the getCpuSchedulingDomains for example?

Reply:
Not sure, the function getCPUswithLoadBalanceDisabled (name change) returns a slice containing  lines which starting with cpu 
`Example: [ cpu0 0 0 0 0 0 0 68186178574807 516377247436 331031573  cpu1 0 0 0 0 0 0 75970491002822 375072790117 330836684]`

And the loop is basically checking this:
1. we have a list of cpus by GU pod (called podCpus)
2. We have a slice containing cpus with load balancing disabled
3.  we loop over podcpus and in the inner loop we loop over the slice and check podCpu are found in the slice.